packages/cardano-services/src/util/validators.ts

```diff
@@ -1,5 +1,41 @@
+import { BuildInfo } from '../Http';
 import { CACHE_TTL_LOWER_LIMIT, CACHE_TTL_UPPER_LIMIT } from '../InMemoryCache';
 import { MissingProgramOption, ProgramOptionDescriptions, ServiceNames } from '../Program';
+import { validate } from 'jsonschema';
+import fs from 'fs';
+
+const buildInfoSchema = {
+  additionalProperties: false,
+  properties: {
+    extra: { type: 'object' },
+    lastModified: { type: 'number' },
+    lastModifiedDate: { type: 'string' },
+    narHash: { type: 'string' },
```

Comment:
Better to nest `narHash`, `path`, and `sourceInfo` under extra properties, since these are specific to nix, or rather, not generic enough to be typed.

Reply:
You mean to nest them in [ServerMetadata](https://github.com/input-output-hk/cardano-js-sdk/blob/88a41a2a0027e9769d42f40a23bf0f851b81b5da/packages/cardano-services/src/Http/types.ts#L15-L25) response type, correct? As the `BUILD_INFO` input all fields are [flat structured](https://preprod-api.dev.lw.iog.io/meta). @rhyslbw

Reply:
In effect, you can remove these properties from our type, and instead expect them to be included as `extra` arguments.

Reply:
Okay, that will force the SRE to pass `BUILD_INFO` value with the new expected structure with `extra` field.
Going to change it now, just not sure wether it should be marked as a breaking change or not? @rhyslbw

Reply:
Good observation. Yes, it's a breaking change

Reply:
4a727e918cae826a61de8d53e82aedc70e2ec4d5 @rhyslbw

Reply:
Extra should remain generic though, as I've suggested here: https://github.com/input-output-hk/cardano-js-sdk/pull/573#discussion_r1069382716

So the only place we see `narHash`, `path`, and `sourceInfo` is in tests, although you could also just use dummy values there.

Reply:
Yeah, for some reason I have paid attention only to the first part (`"to nest narHash, path, and sourceInfo under extra properties"`), and inadvertently missed the second (`"not generic enough to be typed"`). 
It is applied with 3f3931ccdb4fd7b4befe7609b74565678a8f4aee commit @rhyslbw